src/components/ManageGallery/OrganizeCollection/Sidebar/SidebarTokens.tsx

```diff
@@ -45,6 +50,26 @@ export const SidebarTokens = ({
     tokenRefs
   );
 
+  const setSpamPreference = useSetSpamPreference();
+  const setSpamPreferenceForCollection = useCallback(
+    (address: string, isSpam: boolean) => {
+      const groups = groupCollectionsByAddress({ tokens, editModeTokens });
+      const matchingCollection = groups.find((group) => group.address === address);
```

Comment:
correct me if i'm wrong but I think there's some indirection here. 

we have an address, and a list of tokens. we then group the tokens by address into a list. then we search for that group to get the tokens.

I think we can just filter the list of tokens by their contract address.

Reply:
oh nope haha you are 100% right! thanks for catching that, will update